packages/hoppscotch-app/components/realtime/Mqtt.vue

```diff
@@ -175,25 +415,31 @@ const subscriptionState = useReadonlyStream(
   socket.value.subscriptionState$,
   false
 )
+const subscribing = useReadonlyStream(socket.value.subscribing$, false)
 
 const isUrlValid = ref(true)
-const pubTopic = ref("")
 const subTopic = ref("")
-const message = ref("")
+
+// Authorization
+const authTypeOptions = ref<HTMLElement | any>()
+const authType = ref<"none" | "basic" | "token">("none")
+
+// Basic Auth
 const username = ref("")
 const password = ref("")
 
+// Token Auth
+const token = ref("")
+
 let worker: Worker
 
-const canPublish = computed(
-  () =>
-    pubTopic.value !== "" &&
-    message.value !== "" &&
-    connectionState.value === "CONNECTED"
-)
-const canSubscribe = computed(
-  () => subTopic.value !== "" && connectionState.value === "CONNECTED"
-)
+const subscriptionModal = ref(false)
```

Comment:
More descriptive variable names.

```suggestion
const subscriptionModalShown = ref(false)
```